### PR TITLE
refactor: Move some types (not yet exposed to clients) into internal_types

### PR DESCRIPTION
**Pull request by alamb**
# Rationale
The idea is to more clearly separate types which are currently "internal" details from what is 'client' facing as I prepare for system tables in #1013.

*Internal* here means "within the same IOx process" and changing only requires the IOx server itself to deal with differences
*Client facing* means  "exposed via management API or CLI (or soon to be system table #1013)" to clients where changes may require more coordination

Note In this model, "conductor" is a client.

# Changes:
* Move schema to internal types (also removes the arrow dependency)
* Move data (internal format of replicated writes -- internal representation of the WAL)

- [x] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

File: data_types/Cargo.toml [hunk 1]
```diff
@@ -2,13 +2,12 @@
 name = "data_types"
 version = "0.1.0"
 authors = ["pauldix <paul@pauldix.net>"]
+description = "InfluxDB IOx data_types, shared between IOx instances and IOx clients"
 edition = "2018"
+readme = "README.md"
 
 [dependencies] # In alphabetical order
-arrow_deps = { path = "../arrow_deps" }
```
Comment (alamb) [inline data_types/Cargo.toml:8]: Note data_types no longer depends on arrow or flatbuffers

**Review by carols10cents — APPROVED** [review]
A few small suggestions, and someone else should definitely look more closely at which types ended up in which crates. Looks reasonable to me though!

File: data_types/README.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Data Tpes
+
+This crate contains structs that are designed for external consumption (in `influxdb_iox_client` and other "client" facing uses).
```
[the remark this answers is not in the document]
Comment (alamb) [reply data_types/README.md:3]: 🤦

File: internal_types/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,3 @@
+pub mod data;
```
[the remark this answers is not in the document]
Comment (alamb) [reply internal_types/src/lib.rs:1]: Will do -- good idea

File: data_types/README.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+# Data Types
+
+This crate contains types that are designed for external consumption (in `influxdb_iox_client` and other "client" facing uses).
+
+*Client facing* in this case means exposed via management API or CLI and where changing the structs may require addiitonal coordiation / organization with clients.
```
Comment (tustvold) [inline data_types/README.md:5]: coordiation :grinning:

File: internal_types/README.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# Internal Types
+
+This crate contains InfluxDB IOx "internal" types which are shared
+across crates and internally between IOx instances, but not exposed
+externally to clients
+
+*Internal* in this case means that changing the structs is designed not to require addiitonal coordiation / organization with clients.
```
Comment (tustvold) [inline internal_types/README.md:7]: addiitional coordiation - the typos are multiplying!

**Review by tustvold — APPROVED** [review]

**Comment by tustvold** [comment]
I think there might be some other things, like the error logger that could move - but this is great start :+1:

**Comment by alamb** [comment]
> I think there might be some other things, like the error logger that could move 

👍 Good idea. I was even thinking I might be able to move `Predicate` and the `Database` trait... We'll see

File: data_types/README.md (hunk 4 above)
Comment (alamb) [reply data_types/README.md:5]:
```suggestion
*Client facing* in this case means exposed via management API or CLI and where changing the structs may require additional coordination / organization with clients.
```

Thanks -- sorry about the myriad of typos. Time to turn on flyspell I suppose

File: internal_types/README.md (hunk 5 above)
Comment (alamb) [reply internal_types/README.md:7]:
```suggestion
*Internal* in this case means that changing the structs is designed not to require additional coordination / organization with clients.
```

**Pull request merged**